Build error on Windows related to rosbag2_compression

ci.ros2.org Windows builds now fail on rosbag2_compression related to this recently merged PR (https://github.com/ros2/rosbag2/pull/311). It is complaining about logic_error not being a member of 'std', which is strange. Does a more modern version of c++ need to set in CMakeLists.txt?

Example failure:
https://ci.ros2.org/job/ci_windows/9481/

```
15:25:34 [53.157s] C:\ci\ws\src\ros2\rosbag2\rosbag2_compression\src\rosbag2_compression\compression_factory_impl.hpp(37,16): error C2039: 'logic_error': is not a member of 'std' (compiling source file C:\ci\ws\src\ros2\rosbag2\rosbag2_compression\src\rosbag2_compression\compression_factory.cpp) [C:\ci\ws\build\rosbag2_compression\rosbag2_compression.vcxproj]
15:25:34 [53.172s] C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.24.28314\include\string(24): message : see declaration of 'std' (compiling source file C:\ci\ws\src\ros2\rosbag2\rosbag2_compression\src\rosbag2_compression\compression_factory.cpp) [C:\ci\ws\build\rosbag2_compression\rosbag2_compression.vcxproj]
15:25:34 [53.219s] C:\ci\ws\src\ros2\rosbag2\rosbag2_compression\src\rosbag2_compression\compression_factory_impl.hpp(37,1): error C2065: 'logic_error': undeclared identifier (compiling source file C:\ci\ws\src\ros2\rosbag2\rosbag2_compression\src\rosbag2_compression\compression_factory.cpp) [C:\ci\ws\build\rosbag2_compression\rosbag2_compression.vcxproj]
15:25:34 [53.422s] C:\ci\ws\src\ros2\rosbag2\rosbag2_compression\src\rosbag2_compression\compression_factory_impl.hpp(44,16): error C2039: 'logic_error': is not a member of 'std' (compiling source file C:\ci\ws\src\ros2\rosbag2\rosbag2_compression\src\rosbag2_compression\compression_factory.cpp) [C:\ci\ws\build\rosbag2_compression\rosbag2_compression.vcxproj]
15:25:34 [53.438s] C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.24.28314\include\string(24): message : see declaration of 'std' (compiling source file C:\ci\ws\src\ros2\rosbag2\rosbag2_compression\src\rosbag2_compression\compression_factory.cpp) [C:\ci\ws\build\rosbag2_compression\rosbag2_compression.vcxproj]
15:25:34 [53.453s] C:\ci\ws\src\ros2\rosbag2\rosbag2_compression\src\rosbag2_compression\compression_factory_impl.hpp(44,1): error C2065: 'logic_error': undeclared identifier (compiling source file C:\ci\ws\src\ros2\rosbag2\rosbag2_compression\src\rosbag2_compression\compression_factory.cpp) [C:\ci\ws\build\rosbag2_compression\rosbag2_compression.vcxproj]
```

--- a/rosbag2_compression/include/rosbag2_compression/compression_factory.hpp
+++ b/rosbag2_compression/include/rosbag2_compression/compression_factory.hpp
@@ -23,6 +23,12 @@
 #include "compression_options.hpp"
 #include "visibility_control.hpp"
 
+#ifdef _WIN32
+# pragma warning(push)
+# pragma warning(disable:4251)
+#endif
+
+
 namespace rosbag2_compression
 {
 
@@ -63,5 +69,9 @@ private:
 };
 
 }  // namespace rosbag2_compression
+
+#ifdef _WIN32
+# pragma warning(pop)
+#endif
 
 #endif  // ROSBAG2_COMPRESSION__COMPRESSION_FACTORY_HPP_

--- a/rosbag2_compression/src/rosbag2_compression/compression_factory_impl.hpp
+++ b/rosbag2_compression/src/rosbag2_compression/compression_factory_impl.hpp
@@ -16,6 +16,7 @@
 #define ROSBAG2_COMPRESSION__COMPRESSION_FACTORY_IMPL_HPP_
 
 #include <memory>
+#include <stdexcept>
 #include <string>
 
 #include "rosbag2_compression/compression_factory.hpp"

